schema.go

```diff
@@ -652,22 +654,19 @@ func (s *RecordSchema) FingerprintUsing(typ FingerprintType) ([]byte, error) {
 	return s.fingerprinter.FingerprintUsing(typ, s)
 }
 
-// CacheFingerprint returns a special fingerprint of the schema for caching purposes.
+// CacheFingerprint returns unique identity of the schema.
 func (s *RecordSchema) CacheFingerprint() [32]byte {
-	data := make([]any, 0)
-	for _, field := range s.fields {
-		if field.HasDefault() && field.action == FieldSetDefault {
-			data = append(data, field.Name(), field.Default())
-		}
-		if field.action == FieldIgnore {
-			data = append(data, field.Name()+string(FieldIgnore))
+	return s.cacheFingerprinter.CacheFingerprint(s, func() []byte {
+		var defs []any
+		for _, field := range s.fields {
+			if !field.HasDefault() {
+				continue
+			}
+			defs = append(defs, field.Default())
```

Comment:
I saw that you did omit the field name key and FieldIgnore from the Cachefingerprint. I couldn't find a counter-example when they are needed here as they already force changing the canonical fingerprint. So you're right!